transport/grpc/peer.go

```diff
@@ -125,7 +125,7 @@ func (p *grpcPeer) wait() {
 
 func grpcStatusToYARPCStatus(grpcStatus connectivity.State) peer.ConnectionStatus {
 	switch grpcStatus {
-	case connectivity.Ready:
+	case connectivity.Ready, connectivity.Idle:
 		return peer.Available
 	case connectivity.Connecting:
```

Comment:
Please add an integration test (change detector) to trigger this condition and verify client reconnects before peer is marked as available, this should guard yarpc-go against similar breaking changes in grpc-go (likely in future)